src/supautils.c

```diff
@@ -295,36 +297,31 @@ supautils_hook(PROCESS_UTILITY_PARAMS)
 				confirm_reserved_roles(get_rolespec_name(stmt->role), false);
 
 				if (!is_privileged_role()){
+					bypassrls_is_allowed = false;
 					break;
 				}
 
 				/* Check to see if there are any descriptions related to bypassrls. */
 				foreach(option_cell, stmt->options)
 				{
 					DefElem *defel = (DefElem *) lfirst(option_cell);
-
-					if (strcmp(defel->defname, "bypassrls") == 0) {
-						if (bypassrls_option != NULL) {
-							ereport(ERROR,
-									(errcode(ERRCODE_SYNTAX_ERROR),
-									 errmsg("conflicting or redundant options")));
-						}
-						bypassrls_option = defel;
-					}
+					stmt_has_bypassrls = strcmp(defel->defname, "bypassrls") == 0;
 				}
 
-				// Defer setting bypassrls attribute if using `privileged_role`.
-				if (bypassrls_option != NULL) {
-					stmt->options = list_delete_ptr(stmt->options, bypassrls_option);
-				}
+				/* Allow doing BYPASSRLS */
+				if(bypassrls_is_allowed && stmt_has_bypassrls){
+					switch_to_superuser(privileged_extensions_superuser, &already_switched_to_superuser);
```

Comment:
This allows for privilege escalation for the privileged_role, e.g. `alter role r superuser bypassrls;`